samcli/lib/list/resources/resource_mapping_producer.py

```diff
@@ -0,0 +1,149 @@
+
+    def get_translated_dict(self, template_file_dict: Dict[Any, Any]) -> Dict[Any, Any]:
+        """
+        Performs a sam translate on a template and returns the translated template in the form of a dictionary or
+        raises exceptions accordingly
+
+        Parameters
+        ----------
+        template_file_dict: Dict[Any, Any]
+            The template in dictionary format to be translated
+
+        Returns
+        -------
+        response: Dict[Any, Any]
+            The dictionary representing the translated template
+        """
+        try:
+            # Note to check if IAM can be mocked to get around doing a translate without it
+            validator = SamTemplateValidator(
+                template_file_dict, ManagedPolicyLoader(self.iam_client), profile=self.profile, region=self.region
+            )
+            translated_dict: dict
+            translated_dict = yaml.load(validator.get_translated_template_if_valid(), Loader=yaml.FullLoader)
+            return translated_dict
+        except InvalidSamDocumentException as e:
+            raise InvalidSamTemplateException(str(e)) from e
+        except NoRegionFound as no_region_found_e:
+            raise UserException(
+                "AWS Region was not found. Please configure your region through a profile or --region option",
+                wrapped_from=no_region_found_e.__class__.__name__,
+            ) from no_region_found_e
+        except NoCredentialsError as e:
+            raise UserException(
+                "AWS Credentials are required. Please configure your credentials.", wrapped_from=e.__class__.__name__
+            ) from e
+        except ClientError as e:
+            LOG.error("ClientError Exception : %s", str(e))
+            raise SamListUnknownClientError(msg=str(e)) from e
+
+    def produce(self):
+        """
+        Produces the resource data to be printed
+        """
+        sam_template = get_template_data(self.template_file)
+
+        translated_dict = self.get_translated_dict(template_file_dict=sam_template)
+
+        stacks, _ = SamLocalStackProvider.get_stacks(template_file="", template_dictionary=translated_dict)
+        if not stacks or not stacks[0].resources:
+            raise SamListLocalResourcesNotFoundError(msg="No local resources found.")
+        seen_resources = set()
+        resources_list = []
+        if self.stack_name:
+            response = self.get_resources_info()
+            for deployed_resource in response["StackResources"]:
+                resource_data = ResourcesDef(
+                    LogicalResourceId=deployed_resource["LogicalResourceId"],
+                    PhysicalResourceId=deployed_resource["PhysicalResourceId"],
+                )
+                resources_list.append(dataclasses.asdict(resource_data))
+                seen_resources.add(deployed_resource["LogicalResourceId"])
+            for local_resource in stacks[0].resources:
```

Comment:
I get antsy about IndexErrors everytime I see a direct access such as [0]

Reply:
Agreed, I replaced it with a more descriptive constant instead.